### PR TITLE
ENH make sure `eval(repr(o)) == o` and pickling works

### PR DESCRIPTION
**Pull request by beckermr**
This PR ensures that we keep galsim conventions for `eval(repr(o)) == o` and that pickling works.

Follows #58

### CODE REVIEW

**Review by ismael-mendoza — COMMENTED** [review]
@beckermr actually I have a question about how you approached this and I'm open to your suggestions here. 

You replaced all of the relevant `__repr__` with `'jax_galsim.(...)'` rather than just `'galsim.(...)'`. Our intention at the start was to have  have `jax_galsim` as a drop-in replacement for Galsim so that users would do `import jax_galsim as galsim` and then be able to run the same code as in Galsim. I think this is why our `__repr__`s had `'galsim.(...)'` instead of `'jax_galsim.(...)'`. Do you think it's better have explicit module name in the `__repr__` ?  

I don't think this is too important either way but wanted to let you know about our reasoning. 

Also I did notice that the `do_pickle` function needs `galsim.BaseDeviate` which we haven't implemented yet. So right now when you run this function it uses both classic galsim and jax_galsim I think.

**Comment by beckermr** [comment]
I may have gotten myself confused on this one for sure. Let me go back and double check what I did.

**Comment by beckermr** [comment]
OK this one is ready for another look! You are totally right that we want `galsim.` in the `__repr__` strings. I modified the test suite to ensure that pickling and `eval(repr(o))` works properly.

**Review by ismael-mendoza — APPROVED** [review]
LGTM! Just a question on whether you want to remove the commented out print statements in the new test.

File: tests/jax/test_temporary_tests.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        ndarray,
+    )
+    from collections.abc import Hashable
+
+    def func(x):
+        return x
+
+    print("Try pickling ", str(obj1))
+
+    # print('pickled obj1 = ',pickle.dumps(obj1))
```
[the remark this answers is not in the document]
Comment (beckermr) [reply tests/jax/test_temporary_tests.py:119]: These are from the galsim code base that I copied in. I'd prefer to keep them for debugging later.

**Pull request merged**